src/modules/hip/kernel/mel_filter_bank.hpp

```diff
@@ -0,0 +1,164 @@
+                                          RpptMelScaleFormula melFormula,
+                                          Rpp32s numFilter,
+                                          Rpp32f sampleRate,
+                                          bool normalize,
+                                          rpp::Handle& rppHandle)
+{
+    BaseMelScale *melScalePtr;
+    switch (melFormula)
+    {
+        case RpptMelScaleFormula::HTK:
+            melScalePtr = new HtkMelScale;
+            break;
+        case RpptMelScaleFormula::SLANEY:
+        default:
+            melScalePtr = new SlaneyMelScale();
+            break;
+    }
+
+    Rpp32f maxFreq = sampleRate / 2;
+    Rpp32f minFreq = minFreqVal;
+
+    Rpp64f melLow = melScalePtr->hz_to_mel(minFreq);
+    Rpp64f melHigh = melScalePtr->hz_to_mel(maxFreq);
+    Rpp64f melStep = (melHigh - melLow) / (numFilter + 1);
+
+    Rpp32f *scratchMem = rppHandle.GetInitHandle()->mem.mgpu.scratchBufferPinned.floatmem;
+    Rpp32f *normFactors = scratchMem;
+    Rpp32f *weightsDown = scratchMem + numFilter;
+    Rpp32s *intervals = reinterpret_cast<Rpp32s *>(weightsDown + srcDescPtr->h);
+
+    Rpp32s nfft = (srcDescPtr->h - 1) * 2;
+    Rpp32s numBins = nfft / 2 + 1;
+    Rpp64f hzStep = static_cast<Rpp64f>(sampleRate) / nfft;
+    Rpp64f invHzStep = 1.0 / hzStep;
+
+    Rpp32s fftBinStart = std::ceil(minFreq * invHzStep);
+    Rpp32s fftBinEnd = std::ceil(maxFreq * invHzStep);
+    fftBinEnd = std::min(fftBinEnd, numBins);
+
+    std::fill(normFactors, normFactors + numFilter, 1.0f);
+    memset(weightsDown, 0, sizeof(srcDescPtr->h * sizeof(Rpp32f)));
+    std::fill(intervals, intervals + numFilter + 2, -1);
+
+    Rpp32s fftBin = fftBinStart;
+    Rpp64f mel0 = melLow, mel1 = melLow + melStep;
+    Rpp64f fIter = fftBin * hzStep;
+
+    intervals[0] = fftBinStart;
+    intervals[numFilter + 1] = fftBinEnd;
+
+    for (int interval = 1, idX = 0; idX < numFilter + 1; interval++, idX++, mel0 = mel1, mel1 += melStep)
+    {
+        Rpp64f f0 = melScalePtr->mel_to_hz(mel0);
+        Rpp64f f1 = melScalePtr->mel_to_hz(idX == numFilter ? melHigh : mel1);
+        Rpp64f slope = 1.0 / (f1 - f0);
+
+        intervals[interval] = std::ceil(f1 / hzStep);
+
+        if (normalize && idX < numFilter)
+        {
+            Rpp64f f2 = melScalePtr->mel_to_hz(mel1 + melStep);
+            normFactors[idX] = 2.0 / (f2 - f0);
+        }
+
+        for (; fftBin < fftBinEnd && fIter < f1; fftBin++, fIter = fftBin * hzStep) {
+            weightsDown[fftBin] = (f1 - fIter) * slope;
+        }
+    }
+
+    Rpp32s globalThreads_x = dstDescPtr->w;
+    Rpp32s globalThreads_y = dstDescPtr->h;
+    Rpp32s globalThreads_z = dstDescPtr->n;
+
+    hipLaunchKernelGGL(mel_filter_bank_tensor,
+                       dim3(ceil((float)(globalThreads_x + 15) / LOCAL_THREADS_X), ceil((float)(globalThreads_y + 15) / LOCAL_THREADS_Y), ceil((float)globalThreads_z / LOCAL_THREADS_Z)),
```

Comment:
why this + 15 is required?
It is different to our usual kernel launch configuration we have in our other kernel files
https://github.com/ROCm/rpp/blob/develop/src/modules/hip/kernel/brightness.hpp#L193

Reply:
changed